sql/src/test/java/io/crate/analyze/SelectWindowFunctionAnalyzerTest.java

```diff
@@ -140,4 +141,113 @@ public void testOverWithFrameDefinition() {
         assertThat(frameDefinition.start().type(), is(FrameBound.Type.UNBOUNDED_PRECEDING));
         assertThat(frameDefinition.end().type(), is(FrameBound.Type.UNBOUNDED_FOLLOWING));
     }
+
+    @Test
+    public void test_over_references_window_with_order_by() {
+        QueriedSelectRelation<?> analysis = e.analyze(
+            "SELECT AVG(x) OVER w " +
+            "FROM t " +
+            "WINDOW w AS (ORDER BY x)");
+        WindowFunction windowFunction = (WindowFunction) analysis.outputs().get(0);
+        WindowDefinition windowDefinition = windowFunction.windowDefinition();
+        OrderBy orderBy = windowDefinition.orderBy();
+
+        assertThat(orderBy, not(nullValue()));
+        assertThat(orderBy.orderBySymbols().size(), is(1));
+    }
+
+    @Test
+    public void test_over_references_window_with_partition_by() {
+        QueriedSelectRelation<?> analysis = e.analyze(
+            "SELECT AVG(x) OVER w " +
+            "FROM t " +
+            "WINDOW w AS (PARTITION BY x)");
+        WindowFunction windowFunction = (WindowFunction) analysis.outputs().get(0);
+        WindowDefinition windowDefinition = windowFunction.windowDefinition();
+
+        assertThat(windowDefinition.partitions().size(), is(1));
+    }
+
+    @Test
+    public void test_over_with_order_by_references_window_with_partition_by() {
```

Comment:
I'm wondering if we're testing the right cases on this level.

For the analyzer part we could do a test that named windows are generally being resolved, and the tests that make sure all the different properties are there could be unit tests for the `merge` method?